### PR TITLE
E2E test runner can manage dapr component object automatically

### PR DESCRIPTION
**Pull request by youngbupark**
# Description

This PR is to manage dapr component objects(e.g. redis statestore component) automatically, which resolves the effort to install and clean up components during the test.

## Issue reference

#798 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing
* [ ] Extended the documentation
* [ ] Specification has been updated
* [ ] Provided sample for the feature

### CODE REVIEW

File: tests/platforms/kubernetes/component_description.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+package kubernetes
+
+// ComponentDescription holds dapr component description
+type ComponentDescription struct {
+	// Name is the name of dapr component
+	Name string
+	// Type contains component types (<type>.<component_name>)
+	TypeName string
+	// MetaData contains the meta data for dapr component
```
Comment (LMWF) [inline tests/platforms/kubernetes/component_description.go]: "metadata"

File: tests/runner/kube_testplatform.go [hunk 2]
```diff
@@ -43,11 +45,33 @@ func (c *KubeTestPlatform) tearDown() error {
 		log.Errorf("Failed to tearDown AppResources. got: %q", err)
 	}
 
+	if err := c.ComponentResources.tearDown(); err != nil {
+		log.Errorf("Failed to tearDown ComponentResources. got: %q", err)
+	}
+
 	// TODO: clean up kube cluster
```
Comment (shalabhms) [inline tests/runner/kube_testplatform.go]: nit: remove extra line.

File: tests/runner/kube_testplatform.go [hunk 3]
```diff
@@ -43,11 +45,33 @@ func (c *KubeTestPlatform) tearDown() error {
 		log.Errorf("Failed to tearDown AppResources. got: %q", err)
 	}
 
+	if err := c.ComponentResources.tearDown(); err != nil {
+		log.Errorf("Failed to tearDown ComponentResources. got: %q", err)
```
Comment (shalabhms) [inline tests/runner/kube_testplatform.go]: nit: as per Dapr tracing conventional, start tracing with "lowercase"...like "failed to teardown..."

File: tests/runner/kube_testplatform.go [hunk 4]
```diff
@@ -43,11 +45,33 @@ func (c *KubeTestPlatform) tearDown() error {
...
 }
 
+// addComponents adds component to disposable Resource queues
+func (c *KubeTestPlatform) addComponents(comps []kube.ComponentDescription) error {
+	if c.kubeClient == nil {
+		return fmt.Errorf("kubernetes cluster needs to be setup")
+	}
+
+	for _, comp := range comps {
+		c.ComponentResources.Add(kube.NewDaprComponent(c.kubeClient, kube.DaprTestKubeNameSpace, comp))
```
Comment (shalabhms) [inline tests/runner/kube_testplatform.go]: Will kube.NewDaprComponent(c.kubeClient, kube.DaprTestKubeNameSpace, comp) result in error ?

Comment (youngbupark) [reply tests/runner/kube_testplatform.go]: No it will never return error

File: tests/runner/kube_testplatform.go (hunk 2 above)
Comment (youngbupark) [reply tests/runner/kube_testplatform.go]: I added the extra line intentionally...

File: tests/runner/kube_testplatform.go (hunk 3 above)
Comment (youngbupark) [reply tests/runner/kube_testplatform.go]: I corrected the messages

File: tests/platforms/kubernetes/component_description.go (hunk 1 above)
Comment (youngbupark) [reply tests/platforms/kubernetes/component_description.go]: corrected.

**Pull request merged**